### PR TITLE
fix issue #4051

### PR DESCRIPTION
**Pull request by yenkhoon**
This fix the issue #4051 .
function parseRemoteAddr is forcing protocolHTTP and protocolHTTPs to tcp. this causes the bug in the issue #4051 .

I find that the tcp is only needed where net.Dial. So I moved the switch case in to makeHTTPDialer . 

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by melekes** [comment]
Have you tested it? Do you want to try and write unit tests?

**Comment by yenkhoon** [comment]
I have tested it. I just push a simple unit test for it.

File: rpc/lib/client/http_client_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestHTTPClientMakeHTTPDialer(t *testing.T) {
+	remote := []string{"https://foo-bar.com:80", "http://foo-bar.net:80"}
+
+	for _, f := range remote {
+		protocol, address, err := parseRemoteAddr(f)
+		require.NoError(t, err)
+		conn := makeHTTPDialer(f)
```
Comment (melekes) [inline rpc/lib/client/http_client_test.go]:
```suggestion
		dialer := makeHTTPDialer(f)
```

**Comment by melekes** [comment]
Could you please add 

```
- [rpc/lib] \#4051 Fix RPC client, which was previously resolving https protocol to http (@yenkhoon)
```

to CHANGELOG_PENDING.md under BUGFIXES?

**Comment by yenkhoon** [comment]
> Could you please add
> 
> ```
> - [rpc/lib] \#4051 Fix RPC client, which was previously resolving https protocol to http (@yenkhoon)
> ```
> 
> to CHANGELOG_PENDING.md under BUGFIXES?

added

**Comment by yenkhoon** [comment]
> Thanks for contributing to Tendermint 👍

no problem :)

**Pull request merged**